flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherTest.java

```diff
@@ -991,13 +1027,11 @@ public TestingJobManagerRunner createJobManagerRunner(
         final BlockingJobVertex blockingJobVertex = new BlockingJobVertex("testVertex");
         blockingJobVertex.setInvokableClass(NoOpInvokable.class);
         return Tuple2.of(
-                new JobGraph(TEST_JOB_ID, "blockingTestJob", blockingJobVertex), blockingJobVertex);
-    }
-
-    private JobGraph createFailingJobGraph(Exception failureCause) {
-        final FailingJobVertex jobVertex = new FailingJobVertex("Failing JobVertex", failureCause);
-        jobVertex.setInvokableClass(NoOpInvokable.class);
-        return new JobGraph(jobGraph.getJobID(), "Failing JobGraph", jobVertex);
+                JobGraphBuilder.newStreamingJobGraphBuilder()
+                        .setJobId(jobId)
+                        .addJobVertex(blockingJobVertex)
+                        .build(),
+                blockingJobVertex);
     }
 
     private static class FailingJobVertex extends JobVertex {
```

Comment:
Thanks a lot for the various cleanups! I took some notes.

Reply:
I also added a couple of tests which relied on scheduler details. I think before we did not have the necessity to separate tests thoroughly.